packages/service-library/tests/test_archiving_utils.py

```diff
@@ -267,3 +273,46 @@ async def test_unarchive_in_same_dir_as_archive(
         tmp_path,
         None if store_relative_path else dir_with_random_content,
     )
+
+
+@pytest.mark.parametrize(
+    "compress,store_relative_path",
+    itertools.product([True, False], repeat=2),
+)
+async def test_regression_unsupported_characters(
+    tmp_path: Path,
+    compress: bool,
+    store_relative_path: bool,
+) -> None:
+    with tempfile.TemporaryDirectory() as src_dir:
+        dir_to_archive = Path(src_dir)
+        content = "payload"
+        unsupported_file_name = "something\udce6likethis.zip"
```

Comment:
According to [this](https://www.informit.com/articles/article.aspx?p=2274038&seqNum=10), should not surrogate be in pairs? i.e. U+DCE6 is a low surrogate and must be preceded by a high surrogate

Try this
```cmd
$ pip install surrogates
$ python
Python 3.8.10 (default, Jun 15 2021, 10:37:55) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import surrogates
>>> surrogates.decode('\udce6')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".venv/lib/python3.8/site-packages/surrogates/__init__.py", line 73, in decode
    raise DecodeError(_MESSAGE_LOW_SURROGATE_FIRST
surrogates.DecodeError: Low surrogate U+DCE6 not preceded by a high surrogate
>>> surrogates.decode('something\udce6likethis.zip')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".venv/lib/python3.8/site-packages/surrogates/__init__.py", line 73, in decode
    raise DecodeError(_MESSAGE_LOW_SURROGATE_FIRST
surrogates.DecodeError: Low surrogate U+DCE6 not preceded by a high surrogate
```

My point is mainly: is there any kind of faker generator that can produce an spectrum of characters wide enough to test any other issues with zip (which seems to be really picky at so many levels :-) )

Reply:
Sorry for the generated confusion. the issue is with **not decodable  bytes** and not surrogates. Python uses surrogates to encode the aforementioned bytes.